### PR TITLE
[FE] Automate click Edit map in OSM and control OSM opening (web OSM)

### PR DESCRIPTION
**Pull request by amurKontur**
added Edit map in OSM smoke autotests for user and guest

### CODE REVIEW

File: e2e/page-objects/mapPage.ts [hunk 1]
```diff
@@ -112,4 +118,34 @@ export class MapCanvas extends HelperBase {
...
+    expect(zoom).not.toBeNaN();
+    expect(latitude).not.toBeNaN();
+    expect(longitude).not.toBeNaN();
+
+    return {
+      zoomInteger: Math.trunc(zoom),
+      latitudeInteger: Math.trunc(latitude),
+      longitudeInteger: Math.trunc(longitude),
+    };
+  }
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/page-objects/mapPage.ts]: no, it is autotest, not production code!

[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/page-objects/mapPage.ts]: if test fails, it is good. It should fail

File: e2e/page-objects/mapPage.ts [hunk 2]
```diff
@@ -112,4 +114,26 @@ export class MapCanvas extends HelperBase {
...
+    const [zoom, latitude, longitude] = mapData!.split('/').map(Number);
+    expect(zoom).not.toBeNaN();
+    expect(latitude).not.toBeNaN();
+    expect(longitude).not.toBeNaN();
+
+    return {
+      zoomInteger: Math.trunc(zoom),
+      latitudeInteger: Math.trunc(latitude),
+      longitudeInteger: Math.trunc(longitude),
+    };
```
[the remark this answers is not in the document]
Comment (amurKontur) [reply e2e/page-objects/mapPage.ts:137]: it is already in the code
look a bit higher 

expect(mapData).toBeDefined();

[the remark this answers is not in the document]
Comment (amurQA) [reply e2e/page-objects/mapPage.ts:137]: you are wrong, playwright will not execute next code if "expect(mapData).toBeDefined();" has failed

**Review by dqunbp — APPROVED** [review]

**Pull request merged**